modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3691,6 +3694,22 @@ private void _updateTable(
 			}
 
 			if (!values.containsKey(objectField.getName())) {
+				if (objectField.isRequired() &&
+					(workflowAction != WorkflowConstants.ACTION_SAVE_DRAFT)) {
+
+					ObjectEntry objectEntry = fetchObjectEntry(objectEntryId);
```

Comment:
We can create this variable before FOR because the Object Entry would not change.

Reply:
We change the method (fetch) to find, then we would not need to check if it is different from null.